torchdata/dataloader2/graph/utils.py

```diff
@@ -34,6 +30,33 @@ def helper(g) -> None:  # pyre-ignore
     return dps
 
 
+def list_dps(graph: DataPipeGraph, exclude_dps: Optional[Union[DataPipe, List[DataPipe]]] = None) -> List[DataPipe]:
+    r"""
+    Given the graph of DataPipe generated by ``traverse_dps`` function,
+    return DataPipe instances excluding the provided DataPipe and its prior DataPipe graph.
+    """
+    dps: List[DataPipe] = []
+    cache: Set[int] = set()
+
+    if exclude_dps is not None:
+        if isinstance(exclude_dps, (IterDataPipe, MapDataPipe)):
+            cache.add(id(exclude_dps))
+        else:
+            for dp in exclude_dps:  # type: ignore[union-attr]
+                cache.add(id(dp))
+
+    def helper(g) -> None:  # pyre-ignore
+        for dp_id, (dp, src_graph) in g.items():
+            if dp_id not in cache:
+                cache.add(dp_id)
+                dps.append(dp)
```

Comment:
Side note, this doesn't guarantee any order and returns a list. This should suffice for the current use case.

I wonder if people will want a `Dict` (basically just `traverse_dps` without the IDs). To be clear, this seems fine to me for now.

Reply:
If we treat the graph as a tree, there can be in different orders like how do we traverse a tree. Pre-order; post-order, etc. It highly depends on users' preference especially our graph can be more complicated than a tree. I would say let's wait until we have received any use case.